poetry/installation/executor.py

```diff
@@ -295,9 +295,6 @@ def _do_execute_operation(self, operation: "OperationTypes") -> int:
 
         result = getattr(self, f"_execute_{method}")(operation)
 
-        if result != 0:
-            return result
-
```

Comment:
We need to keep this check in place.

The underlying error is that the `pip_install()` function returns a string and not an integer.

We need to encapsulate the `pip_install()` (and `pip_editable_install()`) calls so that the check make in `Executor.run_pip()` (see https://github.com/python-poetry/poetry/blob/master/poetry/installation/executor.py#L320) is also used.

Reply:
Alright, will do - thanks!